File: hotel-front/src/app/dto/BookRoom.ts
export class BookRoom {

  /** id */
  id: number;

  /** 入住日期 */
  checkInDate: string;

  /** 退房日期 */
  checkOutDate: string;

  /** 房间号 */
  roomNo: string;

  /** 姓名 */
  name: string;

  /** 身份证号 */
  idCard: string;

  /** 手机号 */
  phoneNo: string;

  /** 备注 */
  comment: string;

  /** 审核状态 */
  status: string;

  constructor() {
    this.checkInDate = '';
    this.checkOutDate = '';
    this.roomNo = '';
    this.name = '';
    this.idCard = '';
    this.phoneNo = '';
    this.comment = '';
    this.status = '';
  }
}

File: hotel-server/src/main/java/com/hotel/entity/BookRoom.java
package com.hotel.entity;

import lombok.Data;

import javax.persistence.*;

/**
 * BookRoom 用户预定房间
 *
 * @author suwen
 * @date 2020/6/11 上午8:56
 */
@Data
@Entity
@Table(name = "book_room")
public class BookRoom {

  /** id */
  @Id
  @Column
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Integer id;

  /** 入住日期 */
  @Column private String checkInDate;

  /** 退房日期 */
  @Column private String checkOutDate;

  /** 房间号 */
  @Column(length = 20, unique = true)
  private String roomNo;

  /** 姓名 */
  @Column(nullable = false, length = 80)
  private String name;

  /** 身份证号 */
  @Column(nullable = false, length = 18)
  private String idCard;

  /** 手机号 */
  @Column(length = 11)
  private String phoneNo;

  /** 备注 */
  @Column(length = 100)
  private String comment;

  /** 审核状态 */
  @Column(nullable = false)
  private String status;
}
